### ISSUE TITLE
[BUG] 

### ISSUE THREAD

**Issue by Mercerenies** [dictu-lang/Dictu#744]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When given a `Number` argument, `BigInt.new` truncates to the lowest 32 bits before converting to a `BigInt`.

Example:
```
import BigInt;

print(BigInt.new(9999999999).unwrap());
```

### Expected Behavior

I would expect the above code to print `9999999999`. Currently, it prints `1410065407` (which is equal to `9999999999 mod (2^32)`). If an integer number fits exactly into a `double` then I would expect it to be convertible to `BigInt`.

### Steps To Reproduce

Tested at e43580d2. Run the above code snippet.

**Comment by Jason2605** [comment]
Yeah good find! Seems as though I missed this when merging it in, but using the numeric type we are using `bigint_from_int` which is obviously implicitly converting!
